docs/developer/architecture.rst

```diff
@@ -112,18 +112,8 @@ are also plans to expose the Modin DataFrame API as a reduced API set that encom
 the entire pandas/dataframe API. See `experimental features`_ for more information.
```

Comment:
given you've renamed the section this link is now broken

Reply:
Fixed the link and renamed to advanced features, thanks!